packages/smooth_app/lib/data_models/product_image_data.dart

```diff
@@ -1,15 +1,60 @@
-import 'package:openfoodfacts/model/ProductImage.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
 
 class ProductImageData {
   const ProductImageData({
     required this.imageField,
     required this.title,
     required this.buttonText,
-    this.imageUrl,
+    required this.imageDescriptor,
   });
 
   final ImageField imageField;
   final String title;
   final String buttonText;
-  final String? imageUrl;
+  final ImageDescriptor? imageDescriptor;
+
+  String? getImageUrl(ImageSize size) {
+    return imageDescriptor?.createUrlFor(size);
+  }
+}
+
+class ImageDescriptor {
```

Comment:
Not a big fan of that class to be honest.
Please remember that if it's standard code that can be reused, it has probably a good chance to land in off-dart instead of Smoothie.
I think it's the case here, so temporarily just code a static method for "get full image url" in Smoothie, that will move to off-dat within a week.